lib/resolve.js

```diff
@@ -2,15 +2,44 @@
 
 const memoize = require('lodash.memoize')
 
+const addActionsHeader = function (headers) {
+  return Object.assign({'Accept': 'application/vnd.heroku+json; version=3.actions'}, headers || {})
+}
+
+const appAddon = function (heroku, app, id, options = {}) {
+  const headers = addActionsHeader(options.headers)
+  return heroku.post('/actions/addons/resolve', {
+    'headers': headers,
+    'body': {'app': app, 'addon': id, 'addon_service': options.addon_service || null}
+  })
+  .then((addons) => singularize(addons))
+}
+
+exports.appAddon = appAddon
+
 const addonResolver = function (heroku, app, id, options = {}) {
-  const headers = options.headers || {}
+  const headers = addActionsHeader(options.headers)
+
   let getAddon = function (id) {
-    return heroku.get(`/addons/${encodeURIComponent(id)}`, {headers})
+    return heroku.post('/actions/addons/resolve', {
+      'headers': headers,
+      'body': {'app': null, 'addon': id, 'addon_service': options.addon_service || null}
+    })
+    .then((addons) => singularize(addons))
   }
 
-  if (!app || id.indexOf('::') !== -1) return getAddon(id)
-  return heroku.get(`/apps/${app}/addons/${encodeURIComponent(id)}`, {headers})
-    .catch(function (err) { if (err.statusCode === 404) return getAddon(id); else throw err })
+  if (!app || id.indexOf('::') !== -1) {
+    return getAddon(id)
+  } else {
+    return appAddon(heroku, app, id, options)
+    .catch(function (err) {
+      if (err.statusCode === 404) {
```

Comment:
@dickeyxxx fixed in https://github.com/heroku/heroku-cli-addons/pull/57/commits/30ddd55d82c1b27c3aa13ce7bdc4ca63c5d92f08